src/Database/LSMTree/Internal.hs

```diff
@@ -235,18 +255,36 @@ closeSession Session{sessionState} =
 --
 -- For more information, see 'Database.LSMTree.Normal.TableHandle'.
 --
--- NOTE: a table handle is __not__ thread-safe.
-data TableHandle m h = TableHandle {
-    tableSession     :: !(Session m h)
+-- Concurrent access to a table handle is mediated by a multiple-read,
+-- single-write lock. This allows concurrent read access, but only sequential
+-- write access. TODO: more fine-grained concurrent access than a top-level
+-- read-write lock.
```

Comment:
Do I understand this correctly: This comment is not talking about read and write access in the sense of lookups and inserts, but reading and writing the handle open/close state? Every operation except `close` would be considered read access here (but can of course still do modifications to the internal `Mvar`s)?

Reply:
The intended use would be to acquire a write lock for anything that updates any state in the handle, so not just `TableHandleState`. However, I kept the `MVar`s in there so we can implement more fine-grained concurrent access later. I can remove the MVars for now if that makes it clearer

Reply:
That's very coarse-grained, but okay for a first version.